pocld/dht_advertise.cc

```diff
@@ -0,0 +1,108 @@
+/* dht_advertise.cc - part of pocl-daemon that establishes DHT based discovery
+ service to advertise the remote server and its devices.
+
+
+   Copyright (c) 2023-2024 Yashvardhan Agarwal / Tampere University
+
+   Permission is hereby granted, free of charge, to any person obtaining a copy
+   of this software and associated documentation files (the "Software"), to
+   deal in the Software without restriction, including without limitation the
+   rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+   sell copies of the Software, and to permit persons to whom the Software is
+   furnished to do so, subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included in
+   all copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+   FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+   IN THE SOFTWARE.
+*/
+
+#include "dht_advertise.hh"
+#include "pocl_debug.h"
+#include "pocl_runtime_config.h"
+#include <opendht.h>
+
+void SetConfig(dht::DhtRunner::Config *cfg) {
+  dht::crypto::Identity id{};
+
+  auto node_ca = std::make_unique<dht::crypto::Identity>(
+      dht::crypto::generateEcIdentity("DHT Node CA"));
+  id = dht::crypto::generateIdentity("DHT Node", *node_ca);
+
+  cfg->dht_config.node_config.network = 0;
+  cfg->dht_config.node_config.maintain_storage = false;
+  cfg->dht_config.node_config.persist_path = {};
+  cfg->dht_config.node_config.public_stable = false;
+  cfg->dht_config.id = id;
+  cfg->dht_config.cert_cache_all = static_cast<bool>(id.first);
+  cfg->threaded = true;
+  cfg->proxy_server = {};
+  cfg->push_node_id = "dhtnode";
+  cfg->push_token = {};
+  cfg->peer_discovery = false;
+  cfg->peer_publish = false;
+  cfg->dht_config.node_config.max_req_per_sec = -1;
+  cfg->dht_config.node_config.max_peer_req_per_sec = -1;
+  cfg->dht_config.node_config.max_searches = -1;
+  cfg->dht_config.node_config.max_store_size = -1;
+}
+
+/// Called by the daemon to start advertising using DHT using a seprate thread.
+///
+/// \param info contains the serialized data that is to be published on the DHT
+/// network. Data fields shared through info should be known to the client.
+
+void init_dht_advertisement(std::vector<uint8_t> info) {
+
+  dht::DhtRunner node;
```

Comment:
Check the naming caps, this is a C++ file.

Reply:
It is a c++ file now.